### PR TITLE
Timebar refacto/tracks events

### PR DESCRIPTION
**Pull request by nerik**

### CODE REVIEW

File: libs/timebar/src/chartsNew/common/hooks.ts [hunk 1]
```diff
@@ -1,23 +1,85 @@
...
+    const aggregatedChunks = item.chunks.reduce(
+      (currentClusteredEvents: TimebarChartChunk[], currentEvent: TimebarChartChunk) => {
+        const lastClusteredEvent = currentClusteredEvents[currentClusteredEvents.length - 1]
+        const lastType = lastClusteredEvent ? lastClusteredEvent.type : null
+        const lastEnd =
+          lastClusteredEvent && lastClusteredEvent.end
+            ? lastClusteredEvent.end
+            : Number.NEGATIVE_INFINITY
+        const currentEventStartPx = outerScale(currentEvent.start)
+        const lastEndPx = outerScale(lastEnd)
```
Comment (j8seangel) [inline libs/timebar/src/chartsNew/common/hooks.ts:53]: This is returning NaN when `lastEnd` is Number.NEGATIVE_INFINITY I think we are safe using `deltaPx > MIN_DISTANCE_PX_TO_CLUSTER` but worth mentioning

File: libs/timebar/src/chartsNew/tracks-events.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  const {
+    outerScale,
+    outerWidth,
+    graphHeight,
+    tooltipContainer,
+    start,
+    end,
+    outerStart,
+    outerEnd,
+  } = useContext(TimelineContext) as TimelineContextProps
```
Comment (j8seangel) [inline libs/timebar/src/chartsNew/tracks-events.tsx:57]: `as any` and `as TimelineContextProps` not needed anymore after the 2point5 merge

**Review by j8seangel — COMMENTED** [review]
Looks great! Take a look at the first comments and I'd do another round of performance testing later

File: libs/timebar/src/chartsNew/common/hooks.ts [hunk 3]
```diff
@@ -1,23 +1,85 @@
...
+    }
+  })
+}
+
+export const useClusteredChartData = (data: TimebarChartData) => {
+  const { outerStart, outerEnd, outerScale } = useContext(TimelineContext)
+  const delta = +new Date(outerEnd) - +new Date(outerStart)
+  return useMemo(() => {
+    return clusterData(data, outerScale)
+  }, [data, delta])
```
[the remark this answers is not in the document]
Comment (nerik) [reply libs/timebar/src/chartsNew/common/hooks.ts:87]: Not a typo: only change in delta should trigger invalidation of memoized data, but `clusterData` doesn't need it. Will clarify in a comment

File: apps/fishing-map/features/timebar/timebar.selectors.ts [hunk 4]
```diff
@@ -181,102 +266,37 @@ const selectEventsForTracks = createSelector(
...
-
-        return {
-          ...event,
-          color,
-          colorLabels,
-          description,
-          descriptionGeneric,
-        }
+      // Sort events to pick prioritized event on highlight
+      trackEvents.chunks.sort((chunkA, chunkB) => {
```
[the remark this answers is not in the document]
Comment (nerik) [reply apps/fishing-map/features/timebar/timebar.selectors.ts:291]: Re discussion we had about using a Recoil atom to centralize derived data

File: libs/timebar/package.json [hunk 5]
```diff
@@ -25,7 +25,8 @@
     "react-spring": "^8.0.27",
     "resize-observer-polyfill": "^1.5.1",
     "simple-statistics": "^7.7.0",
-    "topojson-client": "^3.0.0"
+    "topojson-client": "^3.0.0",
+    "use-debounce": "^7.0.1"
```
[the remark this answers is not in the document]
Comment (nerik) [reply libs/timebar/package.json:29]: Thanks 👍 
I'm in favour of 2., what do you think?

**Pull request merged**